### ISSUE TITLE
Allow users to pass in additional options

### ISSUE THREAD

**Issue by jneug** [Mc-Zen/tidy#24]
It would be nice to allow arbitrary options to be passed to `#show-module` that are available to themes in `#style-args.options`. This would allow themes to define custom options beyond the ones Tidy offers for further customization of the output. For example, a theme could allow users to pass in a `primary` color, that changes the style of the output.

I see two possible implementations:

1. `#show-module` collects any unknown named arguments into `#style-args.options`.
2. `#show-module` accepts a new parameter `theme-options`.

**Comment by Mc-Zen** [comment]
Hi @jneug ,
so you don't just want to create some theme for one-time personal use only but reuse it and add customization options?

> I see two possible implementations:
> 1. #show-module collects any unknown named arguments into #style-args.options.
> 2. #show-module accepts a new parameter theme-options.

It would definitely be the second option because otherwise, I can't add arguments to `show-module` without potentially breaking user code. 

Can you give a hint of how urgent this is to you? The reason is that I want to redesign how themes work with user-defined types. Then `show-module`, `show-function`, `show-variables`, etc will become types that you can customize with `set` and `show` rules which is much better and more flexible. I don't like to add new features now that will need to be changed later for this redesign. But if this is is important now, let's add it!

**Comment by jneug** [comment]
It's not that urgent. 

This is already possible to an extent by simply post-processing the result of `#parse-module` and adding some keys to the dictionary. 

This is only possible for functions / variables, though, because only those are passed on to the style. I think a native way to configure the style would be nice in the future.

**Comment by Mc-Zen** [comment]
I imagine that styles could just be types as well. They don't produce content but define some settable fields. Then they define `show` rules for all of the components that tidy provides (like functions, variables, type boxes) and here you can just access the settable fields of the style type.

**Comment by Mc-Zen** [comment]
Meanwhile `ultra-style.typ` defines
```typ
#type ultra-style {
  field primary = pink

  show: it => {
    show tidy.function: f => {
      set text(fill: it.primary)
      f.name
      f.description
      ...
    }
    ...
  }
}

```

**Comment by jneug** [comment]
Yes, that would be nice. Currently, theming is somewhat limited to setting show rules in the initial template functions.

I tried to circumvent this by placing `#metadata` in the document and creating a show rule that renders it with the theme. Like this: 

```typst
// In the template file
#let themable(func) = [#metadata(func)<themable>]

#let my-template(theme) = (body) => {
  show <themable>: it => {
    (it.value)(theme)
  }
}

// In the document
#show: my-template((
  primary: red
))

#themable((theme) => text(fill: theme.primary, "Hello, World!"))
```

This works as a temporary workaround, but I fear it might be slow for larger documents.

**Comment by Mc-Zen** [comment]
This is an interesting workaround
